osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1297 @@
+  r["sel_device_support"] = INTEGER(ipmi_mc_sel_device_support(mc));
+  r["sdr_respository_support"] = INTEGER(ipmi_mc_sdr_repository_support(mc));
+  r["sensor_device_support"] = INTEGER(ipmi_mc_sensor_device_support(mc));
+  r["is_active"] = INTEGER(ipmi_mc_is_active(mc));
+
+  unsigned char guid = 0;
+  int rv = ipmi_mc_get_guid(mc, &guid);
+  switch (rv) {
+  case ENOSYS:
+    r["guid"] = "not available";
+    break;
+
+  case 0:
+    r["guid"] = std::string(reinterpret_cast<const char*>(&guid), 16);
+    break;
+
+  default:
+    LOG(ERROR) << "Unexpected error retrieving MC GUID: " << strerror(rv);
+  }
+
+  IPMIClient* c = reinterpret_cast<IPMIClient*>(data);
+  c->insertRowsQueue(r["name"] + "-" + r["device_id"], r);
+}
+
+IPMIClient& IPMIClient::get() {
+  static IPMIClient c;
+  return c;
+}
+
+bool IPMIClient::isUp() {
+  return up_.load();
+}
+
+IPMIClient::~IPMIClient() {
+  up_.store(false);
+}
+
+IPMIClient::IPMIClient()
+    : up_(false), domain_(nullptr), os_hnd_(nullptr, kFreeOSHandle), lanCh_(0) {
+  LOG(WARNING) << "First time initialization of OpenIPMI client.  This could "
+                  "take a few minutes";
+
+  std::unique_ptr<os_handler_t, std::function<void(os_handler_t*&)>> tempHandle(
+      ipmi_posix_setup_os_handler(), kFreeOSHandle);
+  if (tempHandle.get() == nullptr) {
+    LOG(ERROR) << "Could not allocate posix handler with ipmi_smi_setup_con";
+    return;
+  }
+  os_hnd_.swap(tempHandle);
+
+  TLOG << "Setting OpenIPMI log handler";
+  os_hnd_.get()->set_log_handler(os_hnd_.get(), ipmiLoggerCB);
+
+  TLOG << "Initializing OpenIPMI client";
+  int rv = ipmi_init(os_hnd_.get());
+  if (rv != 0) {
+    LOG(ERROR) << "IPMI initialization failed: " << strerror(rv);
+    return;
+  }
+
+  TLOG << "Setting up OpenIPMI connection";
+  // Only support 1 IPMI connection for now..
+  rv = ipmi_smi_setup_con(0, os_hnd_.get(), NULL, &con_);
+  if (rv != 0) {
+    LOG(WARNING) << "Error setting up SMI connection: " << strerror(rv);
+    return;
+  }
+
+  TLOG << "Opening IPMI domain";
+  if (ipmi_open_domain(
+          "", &con_, 1, NULL, NULL, IPMIFullyUpCB, this, NULL, 0, NULL) != 0) {
+    LOG(ERROR) << "Error opening IPMI domain: " << strerror(rv);
+    return;
+  }
+
+  TLOG << "Adding IPMIClient as a dispatcher service";
+  Dispatcher::addService(
+      std::shared_ptr<IPMIClient>(this, [](IPMIClient* c) { c->stop(); }));
+
+  // Set default timeout value.
+  size_t timeout = 360000;
+  auto parser = Config::getParser("ipmi");
+  if (parser != nullptr && parser.get() != nullptr) {
+    const auto& config = parser->getData().get_child(
+        kIPMIConfigParserRootKey, boost::property_tree::ptree("UNEXPECTED"));
+    if (config.get_value("") == "UNEXPECTED") {
+      LOG(WARNING) << "Could not load ipmi configuration root key: "
+                   << kIPMIConfigParserRootKey;
+    }
+
+    timeout = config.get_child("timeout", boost::property_tree::ptree())
+                  .get_value<size_t>(360000);
```

Comment:
Was this supposed to use `timeout` as the default value? `.get_value<size_t>(timeout)`? I believe you can use variables inside of the assignment.